unified-installer/astra-unified-installer.sh

```diff
@@ -208,16 +324,38 @@ load_config_from_file_if_given() {
         add_problem "CONFIG_FILE '$config_file' does not exist" "Given CONFIG_FILE '$config_file' does not exist"
         return 1
     fi
-
-    # shellcheck disable=SC1090
-    source "$config_file"
+    # Store the current env so it can be re-applied after sourcing the config file
```

Comment:
Could we have just enforced either all params through the config file OR all params though args? I.e. just throw and error if someone supplies both? We are making an assumption here that if the user supplies both the one they actually want is in the args, which may or may not be true. We are guessing.

Reply:
It's not really a guess so much as common practice. Almost all CLIs behave that way as far as I know.

Reply:
I don't think a customer would know or assume that. Most CLIs use --option=value type of args which aren't env based. I just thought this would clean the code up a bit and also ensures the customer knows what args they're actually using.

Reply:
Sure, there's definitely the distinction between options/env vars to consider which might make things weird (since env vars can be inherited by the script without the user realizing -- though this can still happen even without this change), but I think it's still better than forcing one or the other.

We can't really force the config file unless we expand the UI instructions, and we can't force CLI args without removing crucial functionality that most customers will want.

The former (forcing the config file) is definitely viable, but unless you're willing to go to PMs about this and get them to reconsider, I don't think it's really negotiable at this stage in the process.